File: InheritanceTest/results/InheritanceTest.py
class BasePrinter:
    def __init__(self):
        self.num_value = 42

    def get_value(self):
        return "Base"
    
    def print_it(self):
        print "BasePrinter: %s" % (self.get_value(), )
    
    def some_base_func(self):
        return self.num_value

class ChildPrinter(BasePrinter):
    def get_value(self):
        return "Child"

class TestClass:
    def get_printer(self, name):
        result = ChildPrinter() if name == "child" else BasePrinter()
        return result
    
    def test_method(self):
        base_printer = self.get_printer("base")
        child_printer = self.get_printer("child")
        base_printer.print_it()
        child_printer.print_it()
        print "%s == %s" % (base_printer.some_base_func(), child_printer.some_base_func(), )
        
        base_p2 = BasePrinter()
        child_p2 = ChildPrinter()
        print "%s == %s" % (base_p2.num_value, child_p2.num_value, )

try:
    TestClass().test_method()
except Exception as err:
    print "Exception: " + err.message

File: InheritanceTest/results/InheritanceTest.php
<?php

class BasePrinter {
    public $num_value = 42;

    function getValue() {
        return "Base";
    }
    
    function printIt() {
        print(("BasePrinter: " . ($this->getValue()) . "") . "\n");
    }
    
    function someBaseFunc() {
        return $this->num_value;
    }
}

class ChildPrinter extends BasePrinter {
    function getValue() {
        return "Child";
    }
}

class TestClass {
    function getPrinter($name) {
        $result = $name == "child" ? new ChildPrinter() : (new BasePrinter());
        return $result;
    }
    
    function testMethod() {
        $base_printer = $this->getPrinter("base");
        $child_printer = $this->getPrinter("child");
        $base_printer->printIt();
        $child_printer->printIt();
        print(("" . ($base_printer->someBaseFunc()) . " == " . ($child_printer->someBaseFunc()) . "") . "\n");
        
        $base_p2 = new BasePrinter();
        $child_p2 = new ChildPrinter();
        print(("" . ($base_p2->num_value) . " == " . ($child_p2->num_value) . "") . "\n");
    }
}

try {
    $c = new TestClass();
    $c->testMethod();
} catch (Exception $err) {
    echo 'Exception: ' . $err->getMessage() . "\n";
}